src/Chainweb/Pact/PactService.hs

```diff
@@ -407,13 +421,54 @@ readAccountGuard
 readAccountGuard pdb account
     = fmap ssnd <$> readCoinAccount pdb account
 
+-- | Rewards table mapping 3-month periods to their rewards
+-- according to the calculated exponential decay over 120 year period
+--
+readRewards
+    :: HasChainGraph v
+    => v -> IO (HashMap BlockHeight P.ParsedDecimal)
+readRewards v = do
+    rs <- LBS.readFile "rewards/miner_rewards.csv"
+    case CSV.decode CSV.NoHeader rs of
+      Left e -> internalError
+        $ "cannot construct miner reward map: "
+        <> sshow e
+      Right vs -> return
+        $ HM.fromList
+        . V.toList
+        . V.map formatRow
+        $ vs
+  where
+    formatRow :: (Word64, Double) -> (BlockHeight, P.ParsedDecimal)
+    formatRow (!a,!b) =
+      let
+        !n = v ^. chainGraph . to (int . size)
+        !m = fromRational $ toRational b
+      in (BlockHeight a, P.ParsedDecimal $ m / n)
```

Comment:
Rewards in the csv are calculated as the total reward across all chains, so we must divide this total by the number of chains in the chainweb graph to get the reward for a single chain